fish_prompt.fish

```diff
@@ -7,13 +7,13 @@ function fish_prompt
     # Save previous exit code
     set --local exit_code $status
     # Prepare line for prompt
-    echo -e -n (__pure_new_line)
-    # Print pre prompt
-    echo -e (__pure_pre_prompt)
+    echo -e -n (__pure_prompt_beginning)
```

Comment:
Do you mind using a single underscore instead of two? No need to be overzealous.

Reply:
Ultimately, @edouard-lopez, you decide, there is no big difference, I just found it to be a little bit prettier and more inline with fish conventions (since this is how fish folks are doing it)

Reply:
@schrodincat Agree with @edouard-lopez. I vaguely recall a conversation in the fish gitter channel about this. IIRC they were saying that double underscores were a historical relict, rather than a hard convention. I could be mixing things up, so if @faho wants to chime in with more information, that would be great.

Reply:
Thanks @schrodincat @jorgebucaran for digging up. I'm interested by @faho feedback as well.

This remind me of what [Python](https://docs.python.org/2/tutorial/classes.html#multiple-inheritance) does:
> a name prefixed with an underscore (e.g. `_spam`) should be treated as a non-public part of the API (whether it is a function, a method or a data member).

> Any identifier of the form `__spam` (at least two leading underscores, at most one trailing underscore) is textually replaced with `_classname__spam`, where `classname` is the current class name with leading underscore(s) stripped.

If we consider `pure` package as a class we will write: `_pure__internal_function`, where:
* we prefix package name with a single underscore ;
* we prefix function name with two underscores.

Reply:
I see little use of imitating Python in this situation—I'd go with `_pkg_func_name`, but it's up to you folks. 👋

Reply:
>IIRC they were saying that double underscores were a historical relict, rather than a hard convention. I could be mixing things up, so if @faho wants to chime in with more information, that would be great.

Two underscores are mostly a convention, one we use too much (e.g. `__fish_git_prompt` shouldn't have them). The only technical thing about it is that some of the completions know to ignore functions prefixed with underscores if the token doesn't start with one.

Personally, I'd have used one underscore, but once you have one convention you should probably stick with it. We've made a half-arsed attempt to move to one-underscore in the past, and that wasn't really worth it. Plus it was half-arsed.

Reply:
Should we still prefix variables if they are private (like `_pure_fresh_session`)?